generate-artifacts.ps1

```diff
@@ -0,0 +1,75 @@
+param(
+    [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+    [string]$ArtifactsStorageAccountName
+)
+
+# default script values
+$rgName = "mate-resources"
+$containerName = "task-artifacts"
+$resourcesTemplateName = "exported-template.json"
+$taskName = "task1"
+$tempFolderPath = "$PWD/temp"
+$artifactsConfigPath = "$PWD/artifacts.json"
+
+# initial validation
+Write-Output "Running initial validation"
+$context = Get-AzContext  
+if ($context)   
+{  
+    Write-Output "Azure Powershell module is installed, account is connected."  
+} else {  
+    throw "Please log in to Azure using Azure Powershell module (run Connect-AzAccount)"
+}  
+
+Write-Output "Checking if storage account exists"
+$storageAccount = Get-AzStorageAccount -Name $ArtifactsStorageAccountName -ResourceGroupName $rgName -ErrorAction SilentlyContinue
+if ($storageAccount) {
+    Write-Output "Storage account found"
+} else { 
+    throw "Unable to find storage account $ArtifactsStorageAccountName in the resource group $rgName. Please make sure, that you specified the correct name of the storage account"
+}
+
+Write-Output "Checking if artifacts storage container exists" 
+$artifactContainer = Get-AzStorageContainer -Name $containerName -Context $storageAccount.Context -ErrorAction SilentlyContinue
```

Comment:
The `Get-AzStorageContainer` cmdlet might not exist. You should use `Get-AzStorageContainer` from the `Az.Storage` module, which requires a storage context. Ensure that the context is correctly retrieved from the storage account.